cmake/modules/hunter_download_server_url.cmake

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2017 NeroBurner
+# All rights reserved.
+
+include(CMakeParseArguments) # cmake_parse_arguments
+
+include(hunter_status_print)
+include(hunter_internal_error)
+include(hunter_test_string_not_empty)
+
+# used gobal variables
+# - HUNTER_DOWNLOAD_SERVER (mandatory)
+#   - URLs are updated to point to the download server
+# - HUNTER_DOWNLOAD_SERVER_BLACKLIST
+#   - blacklist specifies server urls, which are not mapped to the download server
+
+# update download url to point to HUNTER_DOWNLOAD_SERVER
+# - PACKAGE:       name of the package to update the url for
+# - version:       version of the package
+# - URL:           original URL for the package
+# - OUTPUT:        variable to output the updated URL
+function(hunter_download_server_url)
+  set(function_name "hunter_download_server_url")
+  set(function_synopsis "${function_name}(PACKAGE foo VERSION 0.1.2-p3 URL http://original_url.example.com OUTPUT variable)")
+
+  # parse arguments
+  set(one_value_args PACKAGE VERSION URL OUTPUT)
+  cmake_parse_arguments(x "" "${one_value_args}" "" ${ARGV})
+
+  # No free arguments allowed
+  list(LENGTH x_UNPARSED_ARGUMENTS x_len)
+  if(NOT x_len EQUAL 0)
```

Comment:
Add quotes: `if(NOT x_len EQUAL "0")`

Reply:
will do
just curious: why?

Reply:
There can be variable with name `0` defined in user's space, it will leak to this code.

* http://cgold.readthedocs.io/en/latest/tutorials/variables/regular.html#names
* http://cgold.readthedocs.io/en/latest/tutorials/control-structures.html#cmp0054

Reply:
wow, I did not know that `0` could be a variable. Very good to know, thanks!